### ISSUE TITLE
[FINDING] Icons show wrong state with myopenhab (Dev build)

### ISSUE THREAD

**Issue by cribskip** [openhab/openhab-android#665]
Hi,

as posted in https://github.com/openhab/openhab-android/issues/549#issuecomment-357413014, I've compiled and installed HEAD on my mobile. Working great but today I noticed a stange behaviour:

- When connected to myopenhab cloud through mobile data, Icons get shown in their 'default' state (lamps ON, Rollershutters default...).
- Same happens with connection to myopenhab via WIFI.
- When connected locally via WIFI, all is working great.

This may be related to #542 or coming from myopenhab. I didn't investigate.
Would be great if someone could verify this before shipping 2.0.1.

My dev-build's HEAD is up to 136335ee2375483efa59cd9ab32931dc424159b6.

Cheers,
Sascha

**Comment by mueller-ma** [comment]
Can you try it with #562?

**Comment by cribskip** [comment]
I've copied HEAD and applied #562 successfully.

- Icons behaving the same as prior this patch
- Very nice handling and UX when no network is available (graphic and option to try again)

**Comment by mueller-ma** [comment]
Can you capture a log?

**Comment by mueller-ma** [comment]
I tested it in demo mode, all icons there change, when the state changes. #542 fixed the roller shitter icon, which definitly work now.
Myopenhab was recently updated, so it may come from there. Can you open an issue in openhab-cloud?

**Comment by mueller-ma** [comment]
When you clear the icon cache, you will see in the log if the correct icons are fetched.

**Comment by cribskip** [comment]
I've attached the debugger while starting the app. Turns out there's a problem while checking the reachability of my openhab instance. Trying from same phone with browser works just fine.

Maybe it's blocked, habdroid has all permissions allowed - _is there a permission missing?_:
```
at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1273)
                                                                              at libcore.io.BlockGuardOs.connect(BlockGuardOs.java:110)
```

Here's the relevant part of the logcat:
```
02-02 21:04:04.208 23394-23394/org.openhab.habdroid D/ConnectionFactory: Connecting to local URL http://10.0.0.2:8080/
02-02 21:04:04.225 23394-23394/org.openhab.habdroid D/AbstractConnection: Checking reachability of http://10.0.0.2:8080/
02-02 21:04:04.226 23394-23394/org.openhab.habdroid E/AbstractConnection: Error checking if URL is reachable
                                                                          android.os.NetworkOnMainThreadException
                                                                              at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1273)
                                                                              at libcore.io.BlockGuardOs.connect(BlockGuardOs.java:110)
                                                                              at libcore.io.IoBridge.connectErrno(IoBridge.java:154)
                                                                              at libcore.io.IoBridge.connect(IoBridge.java:122)
                                                                              at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:183)
                                                                              at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:452)
                                                                              at java.net.Socket.connect(Socket.java:884)
                                                                              at org.openhab.habdroid.core.connection.AbstractConnection.isReachable(AbstractConnection.java:111)
                                                                              at org.openhab.habdroid.core.connection.ConnectionFactory.getAvailableConnection(ConnectionFactory.java:182)
                                                                              at org.openhab.habdroid.core.connection.ConnectionFactory.getConnection(ConnectionFactory.java:102)
                                                                              at org.openhab.habdroid.ui.OpenHABMainActivity.initializeConnectivity(OpenHABMainActivity.java:350)
                                                                              at org.openhab.habdroid.ui.OpenHABMainActivity.onCreate(OpenHABMainActivity.java:307)
                                                                              at android.app.Activity.performCreate(Activity.java:6357)
                                                                              at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
                                                                              at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2408)
                                                                              at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2515)
                                                                              at android.app.ActivityThread.access$1000(ActivityThread.java:154)
                                                                              at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1379)
                                                                              at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                              at android.os.Looper.loop(Looper.java:157)
                                                                              at android.app.ActivityThread.main(ActivityThread.java:5555)
                                                                              at java.lang.reflect.Method.invoke(Native Method)
                                                                              at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:745)
                                                                              at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:635)
```

The complete logcat is located at this gist:
https://gist.github.com/cribskip/e5115687315187ec32de1a6e884d2f13

**Comment by maniac103** [comment]
Yes, WIP patches can be buggy ;-) That particular problem (NetworkOnMainThreadException) also was discussed there.

**Comment by mueller-ma** [comment]
@cribskip FYI the app isn't called "Habdroid" anymore, just "openHAB Android". However the package name still is org.openhab.habdroid and will probably never be changed.

**Comment by cribskip** [comment]
Thanks @maniac103 for the hint - didn't see this.
I'll close this now as I'm back to HEAD and it's working thus far :)

**Comment by andyzlen** [comment]
Hi,
installed version 2.0.0 (12/16/17) of "openHAB Android" on my Samsung S5 mobile (Marshmallow 6.0.1), since then I faced very same behaviour @cribskip described in the first post:
•	When connected to myopenhab cloud through mobile data connection, icons get shown in their 'default' state (lamps ON, contacts default...).
•	When connected locally via WIFI, all is working fine.
I didn't install dev-build of HEAD - no clue what HEAD means - I did installation via 'Google Play'.
I checked also myopenhab.org, my device is registered, I could pass a message. 
As this issue was closed I have no idea how to deal with.
Is it possibly a configuration problem on my phone or do I have to wait for app[ release 2.0.1](https://github.com/openhab/openhab-android/issues/549)?

Regards, Andy
